File: openapi-test-java-server-spring/testapi/api/PetsApi.java
package testapi.api;

import com.fasterxml.jackson.databind.JsonNode;
import java.lang.Void;
import java.util.List;
import java.util.Optional;
import testapi.model.Error;
import testapi.model.Pet;
import testapi.model.PetCreate;
import testapi.model.PetId;

public interface PetsApi {
  /** Create a pet */
  Response201400<Pet, Error> createPet(PetCreate body);

  /** Delete a pet */
  Void deletePet(
  
    /** The pet ID */
    PetId petId
  );

  /** Get a pet by ID */
  Response200404<Pet, Error> getPet(
  
    /** The pet ID */
    PetId petId
  );

  /** Get pet photo */
  Void getPetPhoto(
  
    /** The pet ID */
    PetId petId
  );

  /** List all pets */
  List<Pet> listPets(
    /** Maximum number of pets to return */
    Optional<Integer> limit,
    /** Filter by status */
    Optional<String> status
  );

  /** Upload a pet photo */
  JsonNode uploadPetPhoto(
    /** The pet ID */
    PetId petId,
    /** Optional caption for the photo */
    String caption,
    /** The photo file to upload */
    Byte[] file
  );
}

File: openapi-test-kotlin-server-spring-client-mp/testapi/api/PetsApi.kt
package testapi.api

import com.fasterxml.jackson.databind.JsonNode
import java.lang.Void
import java.util.Optional
import kotlin.collections.List
import testapi.model.Error
import testapi.model.Pet
import testapi.model.PetCreate

interface PetsApi {
  /** Create a pet */
  fun createPet(body: PetCreate): Response201400<Pet, Error>

  /** Delete a pet */
  fun deletePet(
    /** The pet ID */
    petId: String
  ): Response404Default<Error>

  /** Get a pet by ID */
  fun getPet(
    /** The pet ID */
    petId: String
  ): Response200404<Pet, Error>

  /** Get pet photo */
  fun getPetPhoto(
    /** The pet ID */
    petId: String
  ): Void

  /** List all pets */
  fun listPets(
    /** Maximum number of pets to return */
    limit: Optional<Integer>,
    /** Filter by status */
    status: Optional<String>
  ): List<Pet>

  /** Upload a pet photo */
  fun uploadPetPhoto(
    /** The pet ID */
    petId: String,
    /** Optional caption for the photo */
    caption: String,
    /** The photo file to upload */
    file: Array<Byte>
  ): JsonNode
}